### DISCUSSION TITLE
How to check a user has a password set?

### DISCUSSION THREAD

**Discussion by csalmeida** [supabase/supabase#11704] [Questions]
I'm trying to find a way to query `auth.users` to see if a user has a password set. 
The application I'm working on has a requirement where there should be a field for the user to set a password if they've logged in with an OAuth provider.

I do understand that something like the following wouldn't work because of schema restrictions:

```javascript
const userRecord = supabase.from('users').select('*').eq('email', user?.email)
```

Is the solution to apply a trigger that saves the password status to a `public.users` table instead?
Thanks a lot 🙏

**Comment by GaryAustin1** [comment]
You would write an rpc call with a "security definer" function to access auth.users table and return true/false if password set.  You can use auth.uid() in the function to provide some protection.
